### PR TITLE
Scalable qr

### PR DESCRIPTION
**Pull request by dbo1dyrev**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android
- [ ] n/a

Screenshot 2021-07-29 at 12 25 59

https://user-images.githubusercontent.com/68349436/127467481-85cce475-7bdf-49d6-9e97-17f13b440c62.mov

### CODE REVIEW

File: src/components/themed/Checkbox.js [hunk 1]
```diff
@@ -23,7 +23,10 @@ type Props = {
 export const Checkbox = ({ textStyle, disabled, children, value, onChange, marginRem, paddingRem, ellipsizeMode, numberOfLines }: Props) => {
   const theme = useTheme()
   const styles = getStyles(theme)
-  const spacings = { ...getMarginSpacingStyles(marginRem || 0.5, theme.rem), ...getPaddingSpacingStyles(paddingRem || 1, theme.rem) }
+  const spacings = {
+    ...getMarginSpacingStyles(typeof marginRem === 'undefined' ? 0.5 : marginRem, theme.rem),
```
Comment (dbo1dyrev) [inline src/components/themed/Checkbox.js:27]: `marginRem ?? 0.5`

File: src/components/themed/Checkbox.js [hunk 2]
```diff
@@ -23,7 +23,10 @@ type Props = {
...
+    ...getPaddingSpacingStyles(typeof paddingRem === 'undefined' ? 1 : paddingRem, theme.rem)
```
Comment (dbo1dyrev) [inline src/components/themed/Checkbox.js:28]: `paddingRem ?? 1`

File: src/components/themed/LayoutDependentQrCode.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const styles = getStyles(theme)
+
+  const [qrCodeContainerHeight, setQrCodeContainerHeight] = useState<number>(0)
+
+  const handleQrCodeLayout = (event: any) => {
+    setQrCodeContainerHeight(event.nativeEvent.layout.height)
+  }
+
+  return (
+    <View style={[styles.qrCode, getMarginSpacingStyles(typeof marginRem === 'undefined' ? 2 : marginRem, theme.rem)]} onLayout={handleQrCodeLayout}>
```
Comment (dbo1dyrev) [inline src/components/themed/LayoutDependentQrCode.js:27]: `marginRem ?? 2`

File: src/components/themed/ThemedModal.js [hunk 4]
```diff
@@ -22,14 +22,15 @@ type Props<T> = {
   // Control over the content area:
   flexDirection?: $PropertyType<ViewStyle, 'flexDirection'>,
   justifyContent?: $PropertyType<ViewStyle, 'justifyContent'>,
-  paddingRem?: number[] | number
+  paddingRem?: number[] | number,
+  center?: boolean
```
Comment (dbo1dyrev) [inline src/components/themed/ThemedModal.js:26]: `position?: 'center' | 'bottom'`

File: src/components/themed/ThemedModal.js [hunk 5]
```diff
@@ -42,9 +43,10 @@ export function ThemedModal<T>(props: Props<T>) {
       borderRadius={theme.rem(1)}
       flexDirection={flexDirection}
       justifyContent={justifyContent}
-      margin={[theme.rem(iconRem / 2), 0, 0]}
+      margin={center ? 0 : [theme.rem(iconRem / 2), 0, 0]}
       onCancel={onCancel}
-      padding={packEdges(paddingRem).map(theme.rem)}
+      center={center}
+      padding={center ? 0 : packEdges(paddingRem).map(theme.rem)}
```
Comment (dbo1dyrev) [inline src/components/themed/ThemedModal.js:49]: make it const

File: src/components/themed/ThemedModal.js [hunk 6]
```diff
@@ -42,9 +43,10 @@ export function ThemedModal<T>(props: Props<T>) {
       borderRadius={theme.rem(1)}
       flexDirection={flexDirection}
       justifyContent={justifyContent}
-      margin={[theme.rem(iconRem / 2), 0, 0]}
+      margin={center ? 0 : [theme.rem(iconRem / 2), 0, 0]}
```
Comment (dbo1dyrev) [inline src/components/themed/ThemedModal.js:46]: make it const

File: src/components/themed/ThemedModal.js [hunk 7]
```diff
@@ -42,9 +43,10 @@ export function ThemedModal<T>(props: Props<T>) {
...
       onCancel={onCancel}
-      padding={packEdges(paddingRem).map(theme.rem)}
+      center={center}
```
Comment (dbo1dyrev) [inline src/components/themed/ThemedModal.js:48]: make it const

File: src/components/themed/LayoutDependentQrCode.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import { getMarginSpacingStyles } from '../../util/edges'
+import { QrCode } from '../common/QrCode'
+import { type Theme, cacheStyles, useTheme } from '../services/ThemeContext'
+
+type Props = {
+  data: string,
+  marginRem?: number[] | number
+}
+
+export function LayoutDependentQrCode({ data, marginRem }: Props) {
```
Comment (dbo1dyrev) [inline src/components/themed/LayoutDependentQrCode.js:16]: rename to `QrCode` also combine it with common/QrCode

File: src/components/modals/FullScreenQrCodeModal.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+import { type Theme, cacheStyles, useTheme } from '../services/ThemeContext'
+import { LayoutDependentQrCode } from '../themed/LayoutDependentQrCode'
+import { ThemedModal } from '../themed/ThemedModal'
+
+type Props = {
+  data: string,
+  bridge: AirshipBridge<boolean>
+}
+
+export function FullScreenQrCodeModal({ data, bridge }: Props) {
```
Comment (dbo1dyrev) [inline src/components/modals/FullScreenQrCodeModal.js:16]: `FullScreenQrCodeModal` => `QrCodeModal`

File: src/components/scenes/RequestScene.js [hunk 10]
```diff
@@ -321,11 +319,11 @@ export class RequestComponent extends React.Component<Props, State> {
...
               </InputAccessoryView>
             ) : null}
-            <View style={styles.qrContainer} onLayout={this.handleQrCodeLayout}>
-              {this.state.qrCodeContainerHeight < theme.rem(1) ? null : (
-                <QrCode data={this.state.encodedURI} size={this.state.qrCodeContainerHeight - theme.rem(1)} />
-              )}
-            </View>
+            <TouchableWithoutFeedback onPress={this.handleQrCodePress}>
+              <View style={styles.qrContainer}>
+                <LayoutDependentQrCode data={this.state.encodedURI} />
```
Comment (dbo1dyrev) [inline src/components/scenes/RequestScene.js:324]: `<LayoutDependentQrCode data={this.state.encodedURI} />` move it to const and use it twice

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**